src/Runner.Worker/ActionCommandManager.cs

```diff
@@ -295,8 +295,21 @@ public void ProcessCommand(IExecutionContext context, string line, ActionCommand
             {
                 throw new Exception("Required field 'name' is missing in ##[save-state] command.");
             }
-
-            context.IntraActionState[stateName] = command.Data;
+            // Embedded steps (composite) keep track of the state at the root level
+            if (context.IsEmbedded)
+            {
+                var id = context.EmbeddedId;
+                if (!context.Root.EmbeddedIntraActionState.ContainsKey(id))
+                {
+                    context.Root.EmbeddedIntraActionState[id] = new Dictionary<string, string>();
```

Comment:
should this use `StringComparer.OrdinalIgnoreCase`

Reply:
see `child.IntraActionState = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);`